op-batcher/batcher/config.go

```diff
@@ -1,29 +1,61 @@
 package batcher
 
 import (
+	"context"
+	"math/big"
 	"time"
 
 	"github.com/ethereum/go-ethereum/common"
-	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/log"
 	"github.com/urfave/cli"
 
 	"github.com/ethereum-optimism/optimism/op-batcher/flags"
 	"github.com/ethereum-optimism/optimism/op-batcher/rpc"
+	"github.com/ethereum-optimism/optimism/op-node/eth"
 	"github.com/ethereum-optimism/optimism/op-node/rollup"
-	"github.com/ethereum-optimism/optimism/op-node/sources"
 	oplog "github.com/ethereum-optimism/optimism/op-service/log"
 	opmetrics "github.com/ethereum-optimism/optimism/op-service/metrics"
 	oppprof "github.com/ethereum-optimism/optimism/op-service/pprof"
 	"github.com/ethereum-optimism/optimism/op-service/txmgr"
 	opsigner "github.com/ethereum-optimism/optimism/op-signer/client"
 )
 
+// L1ConfigProvider is a minimal interface that allows the batch submitter to query
+// the L1 chain, and is coalesced into a [txmgr.ETHBackend] for the [TransactionManager].
+//
+//go:generate mockery --name L1ConfigProvider --output ./mocks
+type L1ConfigProvider interface {
```

Comment:
`*ConfigProvider` is an unexpected name for these interfaces. Is it really config we're getting here? Something like `L1Chain` or `L1DataProvider` seems more accurate to me.  It is referenced from the batcher's `Config` but I'd still name it based on what role it serves rather than how it's passed in.

Reply:
Renamed to data providers - that sounds a lot cleaner 🙏